T13900

rshuler opened the pull request:
https://phabricator.endlessm.com/T13900

Commit:
eos-upgrade-eos2-to-eos3: new script for major OS upgrade

Removes all installed legacy app bundles,
points to the latest ostree branches,
and starts the ostree upgrade process.
Requires the user to re-install any desired
apps as flatpaks from the app center
after rebooting into eos3.

Current version supports only single-disk systems.

https://phabricator.endlessm.com/T13900

Commit:
Force uninstalling and removing printers and downloaded drivers

Printers configured in the system will not notice the 32bit > 64bit
as long as they can still find the PPD in /etc/cups/ppd that was
used to configure them in 2.6, which will cause trouble if the
underlying driver has not been automatically migrated to 64bit.

This will be a problem particularly for drivers downloaded from
OpenPrinting.org via eos-config-printer, as there's no easy way
to automatically update them in the system, but could also be
a problem for other printers supported via OS upgrades if they
have 32bit drivers available but not 64bit ones (e.g. all Canon
printers supported by the cnijfilter-common driver), which will
obscurely stop working after the upgrade, even if they will show
up as installed to the user (as they will keep the PPD file).

To prevent the trouble and the confusion, and even though is a
bit of a drastic measure, we will uninstall every printer and
every downloaded driver from the system, so that the user is
forced to manually install the printers again upon reinstallation.

https://phabricator.endlessm.com/T13900

Commit:
eos-upgrade-eos2-to-eos3: add split disk support

By simply creating the flatpak directory on /var/endless-extra
and giving it a copy of the repo config file, flatpaks will be
installed by default on the secondary storage.

https://phabricator.endlessm.com/T13900

Commit:
eos-upgrade-eos2-to-eos3: allow ostree downgrade

Since 2.6.10 was released after the latest eos3 release (3.0.7),
at the moment this appears to ostree as a chronological "downgrade",
so we need to explicitly allow it.

https://phabricator.endlessm.com/T13900

Commit:
eos-upgrade-eos2-to-eos3: install EknServices

This is a flatpak that needs to be installed after booting into eos3.
We leave behind a service in the new eos3 deployment that will run
on rebooting, install the EknServices flatpak, and then delete itself.
Without this, Endless knowledge apps do not register as desktop
search providers.

https://phabricator.endlessm.com/T13900

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -0,0 +1,222 @@
+            ;;
+        -s|--skip)
+            SKIP=true
+            exit 0
```
is this exit intentional?

rshuler replied:
Oh, that's silly.  I must have copied the "-h" below as a starting point.  Yes, should be a "shift" here instead.

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -0,0 +1,222 @@
+    fi
+fi
+
+# Stop any updates in progress
```
does this stop them from respawning, or should we instead stop them with systemctl?

rshuler replied:
I don't know.  Would something other than logging out/back in respawn them?

dsd replied:
Not sure without checking the exact way the services are hooked up, but sometimes systemd can respawn stuff. If these are systemd managed processes then I recommend stopping them with systemd too.

rshuler replied:
Oh, yes, looks like it is important to stop the eos-updater.timer, too.

Looks like what I want is:
```
systemctl stop eos-updater.timer
systemctl stop eos-updater
systemctl stop ostree-daemon
```

though I seem to get an error trying to stop ostree-daemon (with no ostree update in progress):
```
endless@endless:~/Downloads$ sudo systemctl stop ostree-daemon
Failed to stop ostree-daemon.service: Unit ostree-daemon.service not loaded.
```

@dsd @dbnicholson: Can you confirm that `systemctl stop ostree-daemon` is correct?

dbnicholson replied:
You have to kill ostree-daemon since it's spawned via dbus. However, the names changed about 2.4 timeframe. Here's my process for stopping the automatic updater reliably on 2.3:
```
systemctl stop eos-updater.timer
systemctl stop eos-updater.service
killall ostree-daemon
```
On 2.4+:
```
systemctl stop eos-autoupdater.timer
systemctl stop eos-autoupdater.service
killall eos-updater
```

rshuler replied:
Thanks, @dbnicholson.  Since we ensure that this is only run on the latest 2.6, I'll go with what you have for 2.4+.

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -0,0 +1,222 @@
+    grub-install --boot-directory=/sysroot/boot --modules="ext2 part_msdos part_gpt" --directory=/usr/lib/grub/i386-pc $boot_dev
+    sync
+    # Remove the old boot partition
+    sfdisk --dump $boot_dev | grep -v $boot_part | sfdisk --force $boot_dev
```
can you use strace to check the 2.6.10 version of `sfdisk --dump`, make sure it opens the disk as read only?

rshuler replied:
@dsd I'm not sure I understand the concern, as I'm re-writing the partition table anyhow with `sfdisk --force`.  I've tested this and it works -- is there a potential problem that concerns you, or if it works in testing is all okay?

dsd replied:
I saw behaviour in the past with sfdisk opening the device as read-write in similar situations. This causes a race where upon close, udev recreates all the partition devices meaning that the followup sfdisk command will (occasionally) fail. So please do check this or at least send me the strace log from 2.6.10 and I'll look.

rshuler replied:
@dsd: Here you go...  Let me know if anything further is needed...
```
endless@endless:~/Downloads$ ./strace sfdisk --dump
execve("/usr/sbin/sfdisk", ["sfdisk", "--dump"], [/* 44 vars */]) = 0
brk(0)                                  = 0x8774000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap2(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf770d000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=138078, ...}) = 0
mmap2(NULL, 138078, PROT_READ, MAP_PRIVATE, 3, 0) = 0xf76eb000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libfdisk.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0 f\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=391684, ...}) = 0
mmap2(NULL, 394644, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf768a000
mmap2(0xf76e7000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x5d) = 0xf76e7000
mmap2(0xf76ea000, 1428, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf76ea000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libsmartcols.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0P:\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=157652, ...}) = 0
mmap2(NULL, 160596, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf7662000
mmap2(0xf7686000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x24) = 0xf7686000
mmap2(0xf7689000, 852, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf7689000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\3\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0\340\233\1\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0755, st_size=1753800, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf7661000
mmap2(NULL, 1755772, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf74b4000
mmap2(0xf765b000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a7) = 0xf765b000
mmap2(0xf765e000, 10876, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf765e000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libuuid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0p\17\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=17984, ...}) = 0
mmap2(NULL, 20680, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf74ae000
mmap2(0xf74b2000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3) = 0xf74b2000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libblkid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0\240L\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=297420, ...}) = 0
mmap2(NULL, 300380, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf7464000
mmap2(0xf74aa000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x46) = 0xf74aa000
mmap2(0xf74ad000, 1372, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf74ad000
close(3)                                = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf7463000
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf7462000
set_thread_area({entry_number:-1 -> 12, base_addr:0xf7462740, limit:1048575, seg_32bit:1, contents:0, read_exec_only:0, limit_in_pages:1, seg_not_present:0, useable:1}) = 0
mprotect(0xf765b000, 8192, PROT_READ)   = 0
mprotect(0xf74b2000, 4096, PROT_READ)   = 0
mprotect(0xf74aa000, 8192, PROT_READ)   = 0
mprotect(0xf7686000, 8192, PROT_READ)   = 0
mprotect(0xf76e7000, 4096, PROT_READ)   = 0
mprotect(0x805e000, 8192, PROT_READ)    = 0
mprotect(0xf7732000, 4096, PROT_READ)   = 0
munmap(0xf76eb000, 138078)              = 0
ioctl(0, SNDCTL_TMR_TIMEBASE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
brk(0)                                  = 0x8774000
brk(0x8795000)                          = 0x8795000
open("/usr/lib/locale/locale-archive", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=114327312, ...}) = 0
mmap2(NULL, 2097152, PROT_READ, MAP_PRIVATE, 3, 0) = 0xf7262000
mmap2(NULL, 1249280, PROT_READ, MAP_PRIVATE, 3, 0x163) = 0xf7131000
mmap2(NULL, 4096, PROT_READ, MAP_PRIVATE, 3, 0x1256) = 0xf770c000
close(3)                                = 0
ioctl(1, SNDCTL_TMR_TIMEBASE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
openat(AT_FDCWD, "/sysroot/home/endless/.config/terminal-colors.d", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/terminal-colors.d", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=2570, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf770b000
read(3, "# Locale name alias data base.\n#"..., 4096) = 2570
read(3, "", 4096)                       = 0
close(3)                                = 0
munmap(0xf770b000, 4096)                = 0
open("/usr/share/locale/en_US.utf8/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en_US/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en.utf8/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en_US.utf8/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en_US/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en.utf8/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en/LC_MESSAGES/util-linux.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
write(2, "sfdisk: ", 8sfdisk: )                 = 8
write(2, "no disk device specified", 24no disk device specified) = 24
write(2, "\n", 1
)                       = 1
close(1)                                = 0
close(2)                                = 0
exit_group(1)                           = ?
```

dsd replied:
Please trace the output of `sfdisk --dump /dev/foo` where /dev/foo is your disk, sorry if that wasn't clear

rshuler replied:
@dsd: Oops -- sorry about that.  I suppose this will be more helpful...

```
endless@endless:~/Downloads$ sudo ./strace sfdisk --dump /dev/mmcblk0
execve("/usr/sbin/sfdisk", ["sfdisk", "--dump", "/dev/mmcblk0"], [/* 16 vars */]) = 0
brk(0)                                  = 0x9ae2000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap2(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf76f8000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=138078, ...}) = 0
mmap2(NULL, 138078, PROT_READ, MAP_PRIVATE, 3, 0) = 0xf76d6000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libfdisk.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0 f\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=391684, ...}) = 0
mmap2(NULL, 394644, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf7675000
mmap2(0xf76d2000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x5d) = 0xf76d2000
mmap2(0xf76d5000, 1428, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf76d5000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libsmartcols.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0P:\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=157652, ...}) = 0
mmap2(NULL, 160596, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf764d000
mmap2(0xf7671000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x24) = 0xf7671000
mmap2(0xf7674000, 852, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf7674000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\3\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0\340\233\1\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0755, st_size=1753800, ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf764c000
mmap2(NULL, 1755772, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf749f000
mmap2(0xf7646000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a7) = 0xf7646000
mmap2(0xf7649000, 10876, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf7649000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libuuid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0p\17\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=17984, ...}) = 0
mmap2(NULL, 20680, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf7499000
mmap2(0xf749d000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3) = 0xf749d000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/i386-linux-gnu/libblkid.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\1\1\1\0\0\0\0\0\0\0\0\0\3\0\3\0\1\0\0\0\240L\0\0004\0\0\0"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=297420, ...}) = 0
mmap2(NULL, 300380, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xf744f000
mmap2(0xf7495000, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x46) = 0xf7495000
mmap2(0xf7498000, 1372, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xf7498000
close(3)                                = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf744e000
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf744d000
set_thread_area({entry_number:-1 -> 12, base_addr:0xf744d740, limit:1048575, seg_32bit:1, contents:0, read_exec_only:0, limit_in_pages:1, seg_not_present:0, useable:1}) = 0
mprotect(0xf7646000, 8192, PROT_READ)   = 0
mprotect(0xf749d000, 4096, PROT_READ)   = 0
mprotect(0xf7495000, 8192, PROT_READ)   = 0
mprotect(0xf7671000, 8192, PROT_READ)   = 0
mprotect(0xf76d2000, 4096, PROT_READ)   = 0
mprotect(0x805e000, 8192, PROT_READ)    = 0
mprotect(0xf771d000, 4096, PROT_READ)   = 0
munmap(0xf76d6000, 138078)              = 0
ioctl(0, SNDCTL_TMR_TIMEBASE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
brk(0)                                  = 0x9ae2000
brk(0x9b03000)                          = 0x9b03000
open("/usr/lib/locale/locale-archive", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=114327312, ...}) = 0
mmap2(NULL, 2097152, PROT_READ, MAP_PRIVATE, 3, 0) = 0xf724d000
mmap2(NULL, 1249280, PROT_READ, MAP_PRIVATE, 3, 0x163) = 0xf711c000
mmap2(NULL, 4096, PROT_READ, MAP_PRIVATE, 3, 0x1256) = 0xf76f7000
close(3)                                = 0
ioctl(1, SNDCTL_TMR_TIMEBASE or TCGETS, {B38400 opost isig icanon echo ...}) = 0
openat(AT_FDCWD, "/sysroot/home/endless/.config/terminal-colors.d", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/terminal-colors.d", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/dev/mmcblk0", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = 3
fadvise64_64(3, 0, 0, POSIX_FADV_RANDOM) = 0
fstat64(3, {st_mode=S_IFBLK|0660, st_rdev=makedev(179, 0), ...}) = 0
uname({sys="Linux", node="endless", ...}) = 0
ioctl(3, BLKGETSIZE64, 0x9ae5254)       = 0
open("/sys/dev/block/179:0", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = 4
openat(4, "dm/uuid", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = -1 ENOENT (No such file or directory)
close(4)                                = 0
open("/sys/dev/block/179:0", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = 4
fstatat64(4, "partition", 0xffa059dc, 0) = -1 ENOENT (No such file or directory)
openat(4, "dm/uuid", O_RDONLY|O_LARGEFILE|O_CLOEXEC) = -1 ENOENT (No such file or directory)
close(4)                                = 0
ioctl(3, CDROM_GET_CAPABILITY, 0)       = -1 EINVAL (Invalid argument)
ioctl(3, BLKALIGNOFF, 0xffa06b88)       = 0
ioctl(3, BLKIOMIN, 0xffa06b88)          = 0
ioctl(3, BLKIOOPT, 0xffa06b88)          = 0
ioctl(3, BLKPBSZGET, 0xffa06b88)        = 0
ioctl(3, BLKSSZGET, 0x9ae526c)          = 0
ioctl(3, BLKSSZGET, 0xffa06c68)         = 0
uname({sys="Linux", node="endless", ...}) = 0
ioctl(3, BLKGETSIZE64, 0xffa06c20)      = 0
ioctl(3, 0x301, 0xffa06c24)             = 0
_llseek(3, 0, [0], SEEK_SET)            = 0
read(3, "\353c\220\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 512) = 512
fstat64(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 0), ...}) = 0
mmap2(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xf76f6000
write(1, "label: dos\n", 11label: dos
)            = 11
write(1, "label-id: 0x00000000\n", 21label-id: 0x00000000
)  = 21
write(1, "device: /dev/mmcblk0\n", 21device: /dev/mmcblk0
)  = 21
write(1, "unit: sectors\n", 14unit: sectors
)         = 14
write(1, "\n", 1
)                       = 1
write(1, "/dev/mmcblk0p1 : start=        6"..., 74/dev/mmcblk0p1 : start=        6144, size=      188416, type=83, bootable
) = 74
write(1, "/dev/mmcblk0p2 : start=      194"..., 64/dev/mmcblk0p2 : start=      194560, size=    30590976, type=83
) = 64
close(3)                                = 0
close(1)                                = 0
munmap(0xf76f6000, 4096)                = 0
close(2)                                = 0
exit_group(0)                           = ?
```

dsd replied:
opens as read only so that looks good, no changes required, thanks!

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -0,0 +1,222 @@
+rm -rf /var/endless || true
+rm -rf /var/endless-extra/* || true
+
+# Remove any existing subscriptions app data for all users
```
replace with `rm -rf /home/*/.local/share/com.endlessm.subscriptions` ?

rshuler replied:
Oh, yes, that is cleaner/simpler.

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -199,6 +194,36 @@ gpg-verify-summary=false
 url=file:///var/lib/flatpak-external-apps
 EOF
 
+if [[ $(blkid -t LABEL=extra) ]]; then
```
other scripts that look to identify this setup just check for the existence of /dev/disk/by-label/extra

dsd commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -0,0 +1,222 @@
+if [ ! "$(ls -A /sysroot/boot)" ]; then
+    echo "Migrating separate boot partition into main partition"
+    # Get the boot partition (e.g., /dev/sda1 or /dev/mmcblk0p1)
+    boot_part=`mount | grep /boot | awk '{print $1}'`
```
can you use findmnt for this?

rshuler replied:
Oh, looks like `findmnt -T /boot -o SOURCE -n` should work.  I'll give it a try.  Thanks!

Commit:
eos-upgrade-eos2-to-eos3: tabs to spaces

https://phabricator.endlessm.com/T13900

Commit:
eos-upgrade-eos2-to-eos3: address code review comments

Fix the "-s|--skip" parameter.

Clean up the stopping of the update services and make sure
they are not restarted via a timer.

A couple other simplifications to the code with no behavioral changes.

https://phabricator.endlessm.com/T13900

mariospr commented on eos-upgrade-eos2-to-eos3:
```diff
@@ -106,16 +106,17 @@ if ! $FORCE; then
-killall ostree-daemon || true
+# Stop any updates in progress, and prevent the timer from restarting them
+systemctl stop eos-updater.timer || true
+systemctl stop eos-updater.service || true
```
As @dbnicholson suggested in https://github.com/endlessm/eos-customer-support/pull/5/commits/5a229b823ae2b0338cd0483f44290af976943229#r91096639, I think these should be:
```
systemctl stop eos-autoupdater.timer || true
systemctl stop eos-autoupdater.service || true
```
I checked that in my 2.6.10 VM and that seems to be indeed the name of the units, not `eos-updater.<service|timer>`

rshuler replied:
Oh, I had not noticed that difference.  But I'm confused why when I was manually testing, I did not see an error when I tried `sudo systemctl stop eosupdater.timer` / `sudo systemctl stop eos-updater.service`.  Would there be any difference in my case where I started with 2.3.4 and then ostree upgraded to 2.6.10?

dbnicholson replied:
An alias is provided, but I'd suggest using the proper eos-autoupdater names.

rshuler replied:
Okay, thanks.  Updated names are ready for review.

mariospr reviewed (changes requested):
It seems that the name of the system unit for the updater is wrong (it should eos-autoupdater, not eos-updater), but otherwise it looks good to me.
